spring/jta/src/main/java/org/jboss/narayana/quickstart/spring/xa/DummyXAResource.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * JBoss, Home of Professional Open Source
+ * Copyright 2011, Red Hat, Inc. and/or its affiliates,
+ * and individual contributors as indicated by the @author tags.
+ * See the copyright.txt in the distribution for a
+ * full listing of individual contributors.
+ * This copyrighted material is made available to anyone wishing to use,
+ * modify, copy, or redistribute it subject to the terms and conditions
+ * of the GNU Lesser General Public License, v. 2.1.
+ * This program is distributed in the hope that it will be useful, but WITHOUT A
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ * PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more details.
+ * You should have received a copy of the GNU Lesser General Public License,
+ * v.2.1 along with this distribution; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston,
+ * MA  02110-1301, USA.
+ *
+ * (C) 2011,
+ * @author JBoss, by Red Hat.
+ */
+package org.jboss.narayana.quickstart.spring.xa;
+
+import javax.transaction.xa.XAException;
+import javax.transaction.xa.XAResource;
+import javax.transaction.xa.Xid;
+import java.io.Serializable;
+
+public class DummyXAResource implements XAResource, Serializable {
+    static final long serialVersionUID = 1;
+
+    public enum faultType {HALT, EX, NONE}
+
+    private transient faultType fault = faultType.NONE;
+
+    private static int commitRequests = 0;
+
+    private Xid[] recoveryXids;
+
+    public boolean startCalled;
+    public boolean endCalled;
+    public boolean prepareCalled;
+    public boolean commitCalled;
+    public boolean rollbackCalled;
+    public boolean forgetCalled;
+    public boolean recoverCalled;
+
+    public DummyXAResource()
+    {
+        this(faultType.NONE);
+    }
+
+    public DummyXAResource(faultType fault)
+    {
+        this.fault = fault;
+    }
+
+    public void setFault(faultType fault) {
+        this.fault = fault;
+    }
+
+    public void commit(final Xid xid, final boolean arg1) throws XAException
+    {
+        System.out.println("DummyXAResource commit() called, fault: " + fault + " xid: " + xid);
+        commitCalled = true;
+        commitRequests += 1;
+
+        if (fault != null) {
+            if (fault.equals(faultType.EX)) {
```

Comment:
I don't think you use this facility so probably clearest to remove the faultType.EX and this handling